### DISCUSSION TITLE
You exceeded your current quota, please check your plan and billing details.

### DISCUSSION THREAD

**Discussion by GabrielOnDelphi** [AliDehbansiahkarbon/ChatGPTWizard#50] [Q&A]
Hi
I generated a new key from a paid account and I get this message:
"You exceeded your current quota, please check your plan and billing details."

Any ideas?

**Comment by AliDehbansiahkarbon** [comment]
Did you replace the old API key with the new one in the settings form?
![image](https://github.com/AliDehbansiahkarbon/ChatGPTWizard/assets/5601608/65bcbc01-e21f-4e64-b567-5448f7155acd)

**Reply by GabrielOnDelphi** [reply]
Yes. Of course. I am not that stupid :) : ) :)

**Reply by AliDehbansiahkarbon** [reply]
Well, it's not possible, you know better than me.
Could you try Postman to make sure?

Just so you know, the API key is stored in this registry key, double check this one too.
Computer\HKEY_CURRENT_USER\Software\ChatGPTWizard\ChatGPTApiKey

**Reply by GabrielOnDelphi** [reply]
I will make a screenshot of my ChatGPT account and then of the key (can be deleted later so security reasons :)
What is Postman ?

**Reply by AliDehbansiahkarbon** [reply]
Postman is a tool that helps you to test APIs, web services, etc...
Or you can use the "Rest Debugger" in Delphi to test the API of ChatGPT directly.

**Reply by AliDehbansiahkarbon** [reply]
And double check that registry key please.

**Reply by jr-ch** [reply]
I do have the same error. Also a paid account. Same error message. And it's a new installation which means Windows and RadStudio is fresh as well.

**Reply by jr-ch** [reply]
Now as I have read the legal stuff from OpenAI I must say that this ChatGPTWizard is no longer of any useful work. OpenAI has completly changed the way you can access the AI. Now you must pay for every questions through the API (you have to pay for packages of questions). It doesn't matter that you have a paid account.

**Reply by AliDehbansiahkarbon** [reply]
You must be aware Nothing is free in this world and I made it for free in such a world!
In my opinion, it's still very useful and you don't understand the power of the plugin yet.
For your information, this plugin supports two more AI chatbots(WriteSonic and You.com) you can try them maybe cheaper not free forever.

**Reply by jr-ch** [reply]
Ali, I really appreciate your work! But for a single developer with an account at OpenAI it is actually easier/cheaper to use the webpage to communicate with chatgpt.

**Reply by AliDehbansiahkarbon** [reply]
If it's about the money I understand otherwise No, it's not!

It's your choice to use it or not and I respect it but for your information and others the actual idea behind making such a plugin is : 

1- You don't need to quit the IDE to ask the question.
2- You don't need to Open the browser, log in on OpenAI, ask the question, wait for an answer, select, copy, reopen the IDE, and paste the answer in the editor.
3- This plugin is not just the side panel dockable form, it can be used directly inside the editor by asking questions in the editor among your codes.
4- You can select part of your code and try to optimize/debug/complete/rewrite/etc the selected lines by a shortcut or right-click on the selected code.
5- it has a history of your questions with their answers to prevent research or let say save your money!
6- and more...

Good Luck.
